OpenTofu crashes when trying to use multiple Outscale provider aliases with outscale/outscale v1.1.0

---
name: multiple provider aliases crashes with outscale/outscale v1.1.0
about: Even if `tofu plan` works with provider v1.0.1, it crashes with v1.1.0 when using multiple provider aliases
labels: bug, new

---

<!--
Hi there,

Thank you for opening an issue. Please note that we try to keep the this tracker reserved for bug reports and feature requests.
To fix problems, we need clear reproduction cases - we need to be able to see it happen locally. A reproduction case is ideally something we can git-clone or copy-paste and run immediately, without inventing any details or context. 

* A short example can be directly copy-pasteable; longer examples should be in separate git repositories, especially if multiple files are needed
* Please include all needed context. For example, if you figured out that an expression can cause a crash, put the expression in a variable definition or a resource
* Set defaults on (or omit) any variables. The person reproducing it should not need to invent variable settings
* If multiple steps are required, such as running terraform twice, consider scripting it in a simple shell script. Providing a script can be easier than explaining what changes to make to the config between runs.
* Omit any unneeded complexity: remove variables, conditional statements, functions, modules, providers, and resources that are not needed to trigger the bug.
-->

### Terraform Version
<!---
Run `terraform version` to show the version, and paste the result between the ``` marks below.

If you are not running the latest version of Terraform, please try upgrading because your issue may have already been fixed.
-->

```
# tofu version
OpenTofu v1.9.0
on linux_arm64
+ provider registry.opentofu.org/outscale/outscale v1.1.0
```

### Terraform Configuration Files
<!--
Paste the relevant parts of your Terraform configuration between the ``` marks below.

For Terraform configs larger than a few resources, or that involve multiple files, please make a GitHub repository that we can clone, rather than copy-pasting multiple files in here.
-->

main.tf (abstract)
```terraform
module mymodule {
    source = "./mymodule"

    providers = {
        outscale = outscale.customer_account
    }

   settings = {
        name = var.name
       (…)
    }
}
```

provider.tf

```terraform
terraform {
  required_providers {
    outscale = {
      source = "outscale/outscale"
      version = "1.1.0"
    }
  }
}

provider outscale {
    access_key_id  = var.access_key_id
    secret_key_id  = var.secret_key_id
    region         = var.storcloud_region
    endpoints {
        api  = "api.${var.storcloud_region}.outscale.com"
    }
}

provider outscale {
    alias = "customer_account"
    access_key_id  = var.customer_access_key_id
    secret_key_id  = var.customer_secret_key_id
    region         = var.storcloud_region
    endpoints {
        api  = "api.${var.storcloud_region}.outscale.com"
    }
}
```

provider.tf in module _mymodule_

```terraform
terraform {
    required_providers {
        outscale = {
            source  = "outscale/outscale"
            version = "1.1.0"
        }
    }
}
```

### Debug Output
<!--
Full debug output can be obtained by running Terraform with the environment variable `TF_LOG=trace`. Please create a GitHub Gist containing the debug output. Please do _not_ paste the debug output in the issue, since debug output is long.

Debug output may contain sensitive information. Please review it before posting publicly.
-->

Same as crash output.

### Crash Output
<!--
If the console output indicates that Terraform crashed, please share a link to a GitHub Gist containing the output of the `crash.log` file.
-->

Most part is here:

```
# tofu plan --var-file=/secrets/customer-A.tfvars --var-file=/secrets/storcloud.tfvars 

Planning failed. OpenTofu encountered an error while generating this plan.

╷
│ Error: Plugin did not respond
│ 
│   with provider["registry.opentofu.org/outscale/outscale"],
│   on provider.tf line 10, in provider "outscale":
│   10: provider outscale {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ConfigureProvider call. The plugin logs may contain more details.
╵
╷
│ Error: Plugin did not respond
│ 
│   with provider["registry.opentofu.org/outscale/outscale"].customer_account,
│   on provider.tf line 26, in provider "outscale":
│   26: provider outscale {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ConfigureProvider call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-outscale_v1.1.0 plugin:

panic: interface conversion: interface {} is map[string]interface {}, not map[string]string

goroutine 53 [running]:
github.com/outscale/terraform-provider-outscale/outscale.providerConfigureClient(0x4000735400)

```

https://gist.github.com/lpiot/7d0c55e9c2090fd201ef6217eeb63dc9#file-gistfile1-txt

### Expected Behavior
<!--
What should have happened?
-->

`Open Tofu` should be calculating the plan, as it does with `v1.0.0-rc2` and `v1.0.1` of `outscale/outscale` provider.

### Actual Behavior
<!--
What actually happened?
-->

With the provider in `v1.0.1`, it crashes all the way.

### Steps to Reproduce
<!--
Please list the full steps required to reproduce the issue, for example:
1. `terraform init`
2. `terraform apply`
-->

1. `tofu init`
2. `tofu plan --var-file=…`

### Additional Context
<!--
Are there anything atypical about your situation that we should know?
-->

### References
<!--
Are there any other GitHub issues (open or closed) or Pull Requests that should be linked here? 
-->

--- a/outscale/provider.go
+++ b/outscale/provider.go
@@ -236,7 +236,10 @@ func providerConfigureClient(d *schema.ResourceData) (interface{}, error) {
 	}
 	endpointsSet := d.Get("endpoints").(*schema.Set)
 	for _, endpointsSetI := range endpointsSet.List() {
-		endpoints := endpointsSetI.(map[string]string)
+		endpoints := make(map[string]string)
+		for key, value := range endpointsSetI.(map[string]interface{}) {
+			endpoints[key] = value.(string)
+		}
 		for _, endpointServiceName := range endpointServiceNames {
 			config.Endpoints[endpointServiceName] = endpoints[endpointServiceName]
 		}

--- a/outscale/resource_outscale_load_balancer_test.go
+++ b/outscale/resource_outscale_load_balancer_test.go
@@ -44,7 +44,7 @@ func TestAccOthers_LBUBasic(t *testing.T) {
 }
 
 func TestAccOthers_LBUPublicIp(t *testing.T) {
-	t.Skip("will be done soon")
+	t.Skip("Conflict UnlinkPublicIp: will be done soon")
 	t.Parallel()
 	var conf oscgo.LoadBalancer
 
@@ -181,7 +181,7 @@ func testAccOutscaleLBUConfig(r int) string {
 resource "outscale_load_balancer" "barRes" {
 	subregion_names = ["%sa"]
 	load_balancer_name               = "foobar-terraform-elb-%d"
-	
+
 	secured_cookies                  = true
 
 	listeners {
@@ -209,16 +209,16 @@ func testAccOutscaleLBUPublicIpConfig(r int) string {
 	resource "outscale_load_balancer" "barIp" {
 		subregion_names = ["%[1]sa"]
 		load_balancer_name = "foobar-terraform-elb-%[2]d"
-	  
+
 		listeners {
 		  backend_port           = 80
 		  backend_protocol       = "HTTP"
 		  load_balancer_protocol = "HTTP"
 		  load_balancer_port     = 80
 		}
-	  
+
 		public_ip = outscale_public_ip.my_public_ip.public_ip
-	  
+
 		tags {
 		  key = "name"
 		  value = "terraform-internet-facing-lb-with-eip"

--- a/tests/qa_provider_oapi/data/flexible_gpu_catalog/TF-176_flexible_gpu_catalog_datasource_attributes_ok/step1.flexible_gpu_catalog_datasource_attributes_ok.ref
+++ b/tests/qa_provider_oapi/data/flexible_gpu_catalog/TF-176_flexible_gpu_catalog_datasource_attributes_ok/step1.flexible_gpu_catalog_datasource_attributes_ok.ref
@@ -23,43 +23,6 @@
                                 ],
                                 "max_cpu": 35,
                                 "max_ram": 250,
-                                "model_name": "nvidia-a10",
-                                "v_ram": 24000
-                            },
-                            {
-                                "generations": [
-                                    "v7"
-                                ],
-                                "max_cpu": 64,
-                                "max_ram": 512,
-                                "model_name": "nvidia-h100",
-                                "v_ram": 80000
-                            },
-                            {
-                                "generations": [
-                                    "v5"
-                                ],
-                                "max_cpu": 80,
-                                "max_ram": 512,
-                                "model_name": "nvidia-p100",
-                                "v_ram": 16000
-                            },
-                            {
-                                "generations": [
-                                    "v5"
-                                ],
-                                "max_cpu": 80,
-                                "max_ram": 512,
-                                "model_name": "nvidia-p6",
-                                "v_ram": 16000
-                            },
-                            {
-                                "generations": [
-                                    "v5",
-                                    "v6"
-                                ],
-                                "max_cpu": 35,
-                                "max_ram": 250,
                                 "model_name": "nvidia-a100",
                                 "v_ram": 40000
                             },
@@ -77,9 +40,9 @@
                                 "generations": [
                                     "v5"
                                 ],
-                                "max_cpu": 35,
-                                "max_ram": 250,
-                                "model_name": "nvidia-v100",
+                                "max_cpu": 80,
+                                "max_ram": 512,
+                                "model_name": "nvidia-p6",
                                 "v_ram": 16000
                             },
                             {
@@ -96,10 +59,57 @@
                                 "generations": [
                                     "v7"
                                 ],
+                                "max_cpu": 35,
+                                "max_ram": 256,
+                                "model_name": "nvidia-l40",
+                                "v_ram": 48000
+                            },
+                            {
+                                "generations": [
+                                    "v7",
+                                    "v104"
+                                ],
                                 "max_cpu": 130,
                                 "max_ram": 2300,
                                 "model_name": "nvidia-h200",
-                                "v_ram": 141000
+                                "v_ram": 0
+                            },
+                            {
+                                "generations": [
+                                    "v5"
+                                ],
+                                "max_cpu": 80,
+                                "max_ram": 512,
+                                "model_name": "nvidia-p100",
+                                "v_ram": 16000
+                            },
+                            {
+                                "generations": [
+                                    "v7"
+                                ],
+                                "max_cpu": 64,
+                                "max_ram": 512,
+                                "model_name": "nvidia-h100",
+                                "v_ram": 80000
+                            },
+                            {
+                                "generations": [
+                                    "v5"
+                                ],
+                                "max_cpu": 35,
+                                "max_ram": 250,
+                                "model_name": "nvidia-v100",
+                                "v_ram": 16000
+                            },
+                            {
+                                "generations": [
+                                    "v5",
+                                    "v6"
+                                ],
+                                "max_cpu": 35,
+                                "max_ram": 250,
+                                "model_name": "nvidia-a10",
+                                "v_ram": 24000
                             },
                             {
                                 "generations": [
@@ -110,15 +120,6 @@
                                 "max_ram": 256,
                                 "model_name": "nvidia-a100-80",
                                 "v_ram": 80000
-                            },
-                            {
-                                "generations": [
-                                    "v7"
-                                ],
-                                "max_cpu": 35,
-                                "max_ram": 256,
-                                "model_name": "nvidia-l40",
-                                "v_ram": 48000
                             }
                         ],
                         "id": "##id-0##",

